From 977054bc305fc97ec7af1d7271ea0096a1d7d0e1 Mon Sep 17 00:00:00 2001
From: "kidd-kw.chen" <kidd-kw.chen@mediatek.com>
Date: Wed, 3 May 2023 16:08:57 +0800
Subject: [PATCH 1/2] Fix bd compilation issue

Signed-off-by: Julien STEPHAN <jstephan@baylibre.com>
---
 cmake/GlobalConfig.cmake | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake/GlobalConfig.cmake b/cmake/GlobalConfig.cmake
index 9137fa33d..eb472283c 100644
--- a/cmake/GlobalConfig.cmake
+++ b/cmake/GlobalConfig.cmake
@@ -135,13 +135,13 @@ set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules ${CMAKE_MODULE_P
 
 include(CMakeFindDependencyMacro)
 
-if (NOT BUILD_PIPE_ONLY)
+if (BUILD_PIPE_ONLY)
   # cxxopts (Alternative to boost::program_options)
   find_path(CXXOPTS_INCLUDE cxxopts/cxxopts.hpp PATHS third-party NO_CMAKE_FIND_ROOT_PATH)
   include_directories(SYSTEM "${CXXOPTS_INCLUDE}")
 endif()
 
-if (NOT BUILD_PIPE_ONLY)
+if (BUILD_PIPE_ONLY)
   # ghc (Alternative to boost::filesystem)
   find_path(GHC_INCLUDE ghc/filesystem.hpp PATHS third-party NO_CMAKE_FIND_ROOT_PATH)
   include_directories(SYSTEM "${GHC_INCLUDE}")
-- 
2.18.0

