From c86ce777fadba3815500266e1e28c6023130106a Mon Sep 17 00:00:00 2001
From: Julien STEPHAN <jstephan@baylibre.com>
Date: Tue, 25 May 2021 14:26:32 +0200
Subject: [PATCH] fix include directive for CL/cl_ext.h file

Signed-off-by: Julien STEPHAN <jstephan@baylibre.com>
---
 src/backends/cl/ClImportTensorHandle.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/backends/cl/ClImportTensorHandle.hpp b/src/backends/cl/ClImportTensorHandle.hpp
index ddd20fc49..752f79332 100644
--- a/src/backends/cl/ClImportTensorHandle.hpp
+++ b/src/backends/cl/ClImportTensorHandle.hpp
@@ -19,7 +19,7 @@
 #include <arm_compute/core/TensorShape.h>
 #include <arm_compute/core/Coordinates.h>
 
-#include <include/CL/cl_ext.h>
+#include <CL/cl_ext.h>
 #include <arm_compute/core/CL/CLKernelLibrary.h>
 
 namespace armnn
