From 8f512f42d5aff262b9a5e4889ef9b1f8b1cef894 Mon Sep 17 00:00:00 2001
From: Julien STEPHAN <jstephan@baylibre.com>
Date: Mon, 30 Nov 2020 14:22:05 +0000
Subject: [PATCH] search for system opencl header

Signed-off-by: Julien STEPHAN <jstephan@baylibre.com>
---
 cmake/GlobalConfig.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cmake/GlobalConfig.cmake b/cmake/GlobalConfig.cmake
index b093e5d52..70fb1a34e 100644
--- a/cmake/GlobalConfig.cmake
+++ b/cmake/GlobalConfig.cmake
@@ -318,6 +318,7 @@ if(ARMCOMPUTECL)
               PATHS ${ARMCOMPUTE_ROOT}/include
               NO_DEFAULT_PATH NO_CMAKE_FIND_ROOT_PATH)
 
+    find_path(OPENCL_INCLUDE CL/cl2.hpp)
     # Link against libOpenCL in opencl-1.2-stubs, but don't search there at runtime
     link_libraries(-L${ARMCOMPUTE_BUILD_DIR}/opencl-1.2-stubs)
     set(OPENCL_LIBRARIES OpenCL)
-- 
2.17.1

