From 650836ecf60bda499d0485d649da3c44ba58a396 Mon Sep 17 00:00:00 2001
From: Julien STEPHAN <jstephan@baylibre.com>
Date: Tue, 26 Jan 2021 14:33:49 +0000
Subject: [PATCH 3/4] xtensa-ann: add Makefile

Signed-off-by: Julien STEPHAN <jstephan@baylibre.com>
---
 Makefile | 100 +++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 100 insertions(+)
 create mode 100644 Makefile

diff --git a/Makefile b/Makefile
new file mode 100644
index 0000000..1a5bca4
--- /dev/null
+++ b/Makefile
@@ -0,0 +1,100 @@
+DESTDIR ?=
+LIBDIR ?= /usr/lib
+SYSROOT ?=
+
+SRC := \
+        xtensa_driver/XtensaDriver.cpp \
+        xtensa_driver/XtensaExecutor.cpp \
+        xtensa_driver/operations/src/ann_v1_1_prepare.cpp \
+        xtensa_driver/operations/src/common.cpp \
+        xtensa_driver/operations/src/conv.cpp \
+        xtensa_driver/operations/src/deconv.cpp \
+        xtensa_driver/operations/src/pooling.cpp \
+        xtensa_driver/operations/src/RELU.cpp \
+        xtensa_driver/operations/src/softmax.cpp \
+        xtensa_driver/operations/src/logistic.cpp \
+        xtensa_driver/operations/src/concat.cpp \
+        xtensa_driver/operations/src/l2norm.cpp \
+        xtensa_driver/operations/src/eltwiseOp.cpp \
+        xtensa_driver/operations/src/lookup.cpp \
+        xtensa_driver/operations/src/depthnspace.cpp \
+        xtensa_driver/operations/src/split.cpp \
+        xtensa_driver/operations/src/oem.cpp \
+        xtensa_driver/operations/src/crop.cpp \
+        xtensa_driver/operations/src/transpose4D.cpp \
+        xtensa_driver/operations/src/pad.cpp \
+        xtensa_driver/operations/src/mean4D.cpp \
+        xtensa_driver/operations/src/batchnspace.cpp \
+        xtensa_driver/operations/src/slice.cpp \
+        xtensa_driver/operations/src/squeeze.cpp \
+        xtensa_driver/operations/src/dilated_conv.cpp \
+        xtensa_driver/operations/src/LSTM.cpp \
+        xtensa_driver/operations/src/QuantizedLSTM.cpp \
+        xtensa_driver/operations/src/PRELU.cpp \
+        xtensa_driver/operations/src/tanh.cpp \
+        xtensa_driver/operations/src/argMinMax.cpp \
+        xtensa_driver/operations/src/channelshuffle.cpp \
+        xtensa_driver/operations/src/grouped_conv.cpp \
+        xtensa_driver/operations/src/transpose_conv.cpp \
+        xtensa_driver/operations/src/topkv2.cpp \
+        xtensa_driver/operations/src/cast.cpp \
+        xtensa_driver/operations/src/gather.cpp \
+        xtensa_driver/operations/src/logsoftmax.cpp \
+        xtensa_driver/operations/src/heatMap.cpp \
+        xtensa_driver/operations/src/reduce.cpp \
+        xtensa_driver/operations/src/reduce4D.cpp \
+        xtensa_driver/operations/src/generate_proposals.cpp \
+        xtensa_driver/operations/src/select.cpp \
+        xtensa_driver/operations/src/hardSwish.cpp \
+        xtensa_driver/operations/src/detection_pp.cpp \
+        xtensa_driver/operations/src/bbox_transform.cpp \
+        xtensa_driver/operations/src/boxNMSlimit.cpp \
+        xtensa_driver/operations/src/tile.cpp \
+        xtensa_driver/operations/src/roiAlign.cpp \
+        xtensa_driver/operations/src/quantize.cpp \
+        xtensa_driver/operations/src/dequantize.cpp \
+        xtensa_driver/operations/src/resizeNearNeighbor.cpp \
+        xtensa_driver/operations/src/resizeBilinear.cpp \
+        xtensa_driver/operations/src/tensorArithmetic.cpp \
+        xtensa_driver/operations/src/conversion.cpp \
+        xtensa_driver/operations/src/eltwiseCompare.cpp \
+        xtensa_driver/operations/src/enhancedLSTM.cpp \
+        xtensa_driver/operations/src/logicalOps.cpp \
+        xtensa_driver/operations/src/logicalNot.cpp \
+        xtensa_driver/operations/src/div.cpp \
+        xtensa_driver/operations/src/control.cpp \
+        xtensa_driver/xicnna/libxi_ref/src/cnn_deconvolveA_c.c \
+        xtensa_driver/xicnna/libxi_ref/src/cnn_ref/ann_helper.c \
+        xtensa_driver/xicnna/libxi_ref/src/cnn_ref/cnn_misc_ref.c \
+        xtensa_driver/xicnna/libxi_ref/src/misc_ref.c \
+		xtensa_driver/nn_driver.cpp \
+
+TMP_OBJECTS = $(SRC:.cpp=.o)
+OBJECTS += $(TMP_OBJECTS:.c=.o)
+LIBNAME := libvendor-nn-hal.so.1.0
+SONAME  := libvendor-nn-hal.so.1
+SOLINK  := libvendor-nn-hal.so
+
+DEFINES =  -D __ANDROID_API__=30 -D __ANDROID_API_O_MR1__=27 -D __ANDROID_API_P__=28 -D __ANDROID_API_Q__=29 -D __ANDROID_API_R__=30 -D __ANDROID_API_FUTURE__
+
+DEFINES += -DXCHAL_VISION_TYPE=6 -DINCLUDE_XI_CNN -DALLOW_INPUT_COEFFICIENTS=false -DHAVE_ATOMIC -DDISABLE_SOFTMAX_FP16=true -DDISABLE_LOG_SOFTMAX_FP16=true -DDISABLE_ROI_ALIGN_QUANT=false -DDISABLE_HEATMAP_MAX_KEYPOINT_QUANT=false -DDISABLE_BOX_WITH_NMS_LIMIT_QUANT=false -DDISABLE_GENERATE_PROPOSALS_QUANT=true -DDISABLE_AXIS_ALIGNED_BBOX_TRANSFORM_QUANT=false -DDISABLE_L2_NORM=false -DDISABLE_RESIZE=true -DDISABLE_EMBEDDING_LOOKUP=false -DDISABLE_HASHTABLE_LOOKUP=false -DHAVE___U64 -DHAVE___U32 -D__ANDROID__
+
+INCLUDES = -I $(AOSP_INCLUDE) -I $(NNAPI_INCLUDE) -I $(NNAPI_INCLUDE)/nn/common/include -I $(NNAPI_INCLUDE)/nn/runtime/include -I xtensa_driver  -I $(LIBAPU_INCLUDE) -I xtensa_driver/operations/include/ -I xtensa_driver/xicnna/libxi/include/ -I xtensa_driver/xicnna/libxi_ref/include/
+
+CFLAGS += -Wall -Werror -Wextra -fPIC -fvisibility=default -Wno-inconsistent-missing-override -Wno-sign-compare -Wl,--no-as-needed -Wl,--no-undefined -fexceptions -Wno-format-security -Wno-unused-parameter -Wno-unused-variable -Wno-unneeded-internal-declaration -Wno-unused-function -Wno-reorder-init-list -Wno-tautological-overlap-compare -Wno-c99-designator -ftrivial-auto-var-init=zero -enable-trivial-auto-var-init-zero-knowing-it-will-be-removed-from-clang -Wno-unused-but-set-variable -Wno-unused-but-set-parameter
+
+CXXFLAGS += -std=c++17
+
+CPPFLAGS += $(DEFINES) $(INCLUDES) $(CFLAGS)
+
+$(LIBNAME): $(OBJECTS)
+	$(CC) -Wl,--whole-archive $(SYSROOT)/$(LIBDIR)/libapu.a $(SYSROOT)/$(LIBDIR)/libxrp.a  -Wl,--no-whole-archive -shared -Wl,-soname,$(SONAME) -o $@ $^ -lapu -lxrp -lpthread -lrt -ltensorflowlite -ldrm -ldrm_apu
+
+install: $(LIBNAME)
+	mkdir -p $(DESTDIR)/$(LIBDIR)/
+	cp $(LIBNAME) $(DESTDIR)/$(LIBDIR)/
+	ln -sf $(LIBNAME) $(DESTDIR)/$(LIBDIR)/$(SONAME)
+	ln -sf $(LIBNAME) $(DESTDIR)/$(LIBDIR)/$(SOLINK)
+
+clean:
+	rm -rf $(OBJECTS) $(LIBNAME)
-- 
2.32.0

