From cd8d4b700db4f0391dfb4f55f74222458b55cc89 Mon Sep 17 00:00:00 2001
From: Julien STEPHAN <jstephan@baylibre.com>
Date: Fri, 25 Sep 2020 11:15:59 +0200
Subject: [PATCH] libbase: add Makefile

Signed-off-by: Julien STEPHAN <jstephan@baylibre.com>
---
 Makefile | 10 ++++++++++
 1 file changed, 10 insertions(+)
 create mode 100644 Makefile

diff --git a/aosp_libs/libbase/Makefile b/aosp_libs/libbase/Makefile
new file mode 100644
index 0000000..ffab276
--- /dev/null
+++ b/aosp_libs/libbase/Makefile
@@ -0,0 +1,10 @@
+SRC := chrono_utils.cpp errors_unix.cpp file.cpp liblog_symbols.cpp logging.cpp mapped_file.cpp parsebool.cpp process.cpp properties.cpp stringprintf.cpp strings.cpp test_utils.cpp threads.cpp
+
+OBJECTS := $(SRC:.cpp=.o)
+TARGET := $(OUT)/libbase.a
+
+$(TARGET): $(OBJECTS)
+	$(AR) rcs $@ $^
+
+clean:
+	rm -rf $(OBJECTS) $(TARGET)
-- 
2.28.0

