From 98179ee7dfe55890a3d27ed78afb79dce04e8398 Mon Sep 17 00:00:00 2001
From: Julien STEPHAN <jstephan@baylibre.com>
Date: Fri, 4 Sep 2020 14:57:28 +0200
Subject: [PATCH] liblog: add Makefile

Signed-off-by: Julien STEPHAN <jstephan@baylibre.com>
---
 Makefile | 10 ++++++++++
 1 file changed, 10 insertions(+)
 create mode 100644 Makefile

diff --git a/aosp_libs/logging/liblog/Makefile b/aosp_libs/logging/liblog/Makefile
new file mode 100644
index 0000000..d567edb
--- /dev/null
+++ b/aosp_libs/logging/liblog/Makefile
@@ -0,0 +1,10 @@
+SRC := log_event_list.cpp log_event_write.cpp log_time.cpp logger_write.cpp logprint.cpp properties.cpp
+
+OBJECTS := $(SRC:.cpp=.o)
+TARGET := $(OUT)/liblog.a
+
+$(TARGET): $(OBJECTS)
+	$(AR) rcs $@ $^
+
+clean:
+	rm -rf $(OBJECTS) $(TARGET)
--
2.28.0

