From f65b355c3f06ca3533573d63e7f4fb0e243293c7 Mon Sep 17 00:00:00 2001
From: Julien STEPHAN <jstephan@baylibre.com>
Date: Fri, 28 Aug 2020 15:43:58 +0200
Subject: [PATCH 2/2] libcutils: add Makefile

Signed-off-by: Julien STEPHAN <jstephan@baylibre.com>
---
 Makefile | 10 ++++++++++
 1 file changed, 10 insertions(+)
 create mode 100644 Makefile

diff --git a/aosp_libs/libcutils/Makefile b/aosp_libs/libcutils/Makefile
new file mode 100644
index 0000000..2275eca
--- /dev/null
+++ b/aosp_libs/libcutils/Makefile
@@ -0,0 +1,10 @@
+SRC := config_utils.cpp hashmap.cpp load_file.cpp native_handle.cpp threads.cpp trace-host.cpp $(NNAPI_SRC)/libcutils/ashmem_chromeos.cpp
+
+OBJECTS := $(SRC:.cpp=.o)
+TARGET := $(OUT)/libcutils.a
+
+$(TARGET): $(OBJECTS)
+	$(AR) rcs $@ $^
+
+clean:
+	rm -rf $(OBJECTS) $(TARGET)
--
2.30.0

