From 682cd575c631036514a600ae4422d9099624eaa1 Mon Sep 17 00:00:00 2001
From: Julien STEPHAN <jstephan@baylibre.com>
Date: Fri, 4 Sep 2020 15:04:34 +0200
Subject: [PATCH 2/2] libutils: add Makefile

Signed-off-by: Julien STEPHAN <jstephan@baylibre.com>
---
 Makefile | 10 ++++++++++
 1 file changed, 10 insertions(+)
 create mode 100644 Makefile

diff --git a/aosp_libs/libutils/Makefile b/aosp_libs/libutils/Makefile
new file mode 100644
index 0000000..d9af799
--- /dev/null
+++ b/aosp_libs/libutils/Makefile
@@ -0,0 +1,10 @@
+SRC :=  Errors.cpp JenkinsHash.cpp NativeHandle.cpp RefBase.cpp SharedBuffer.cpp StrongPointer.cpp SystemClock.cpp Timers.cpp VectorImpl.cpp misc.cpp
+
+OBJECTS := $(SRC:.cpp=.o)
+TARGET := $(OUT)/libutils.a
+
+$(TARGET): $(OBJECTS)
+	$(AR) rcs $@ $^
+
+clean:
+	rm -rf $(OBJECTS) $(TARGET)
-- 
2.28.0

