From 9d541250814d59241cb979bbc138757b297b46de Mon Sep 17 00:00:00 2001
From: Julien STEPHAN <jstephan@baylibre.com>
Date: Fri, 28 Aug 2020 15:45:53 +0200
Subject: [PATCH 3/3] libfmq: add Makefile

Signed-off-by: Julien STEPHAN <jstephan@baylibre.com>
---
 Makefile | 12 ++++++++++++
 1 file changed, 12 insertions(+)
 create mode 100644 Makefile

diff --git a/aosp_libs/libfmq/Makefile b/aosp_libs/libfmq/Makefile
new file mode 100644
index 0000000..000ad39
--- /dev/null
+++ b/aosp_libs/libfmq/Makefile
@@ -0,0 +1,12 @@
+SRC := EventFlag.cpp FmqInternal.cpp
+
+CPPFLAGS += -I include
+
+OBJECTS := $(SRC:.cpp=.o)
+TARGET := $(OUT)/libfmq.a
+
+$(TARGET): $(OBJECTS)
+	$(AR) rcs $@ $^
+
+clean:
+	rm -rf $(OBJECTS) $(TARGET)
--
2.30.0

