From 1f2fc913bb078b008a09c90cc8f872ab8c015bc2 Mon Sep 17 00:00:00 2001
From: Julien STEPHAN <jstephan@baylibre.com>
Date: Tue, 26 Jan 2021 11:30:43 +0100
Subject: [PATCH 6/6] libnnapi-support: Makefile: install missing hidl header

Signed-off-by: Julien STEPHAN <jstephan@baylibre.com>
---
 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index 3aaa817aca..39c48a1745 100644
--- a/Makefile
+++ b/Makefile
@@ -52,6 +52,7 @@ install:
 	cp -r $(LIBFMQ_SRC)/include/* $(DESTDIR)/$(INCLUDEDIR)/aosp/
 	cp -r $(LIBHIDL_SRC)/base/include/* $(DESTDIR)/$(INCLUDEDIR)/aosp/
 	cp -r $(LIBHIDL_SRC)/libhidlmemory/include/* $(DESTDIR)/$(INCLUDEDIR)/aosp/
+	cp -r $(LIBHIDL_SRC)/transport/include/hidl/ $(DESTDIR)/$(INCLUDEDIR)/aosp/
 	mkdir -p ${DESTDIR}/$(INCLUDEDIR)/aosp/android
 	cp $(LIBNATIVE_SRC)/include/android/sharedmem.h $(DESTDIR)/$(INCLUDEDIR)/aosp/android/
 	mkdir -p ${DESTDIR}/$(INCLUDEDIR)/aosp/ui
-- 
2.30.0

