diff --git a/nnapi/hardware_buffer/AHardwareBuffer.cpp b/nnapi/hardware_buffer/AHardwareBuffer.cpp
index bb53b614616..6c4e6331ee2 100644
--- a/nnapi/hardware_buffer/AHardwareBuffer.cpp
+++ b/nnapi/hardware_buffer/AHardwareBuffer.cpp
@@ -16,6 +16,7 @@
 #include <android/log.h>
 #include <vndk/hardware_buffer.h>

+extern "C" {
 int AHardwareBuffer_allocate(const AHardwareBuffer_Desc* /*desc*/,
                              AHardwareBuffer** /*outBuffer*/) {
   return 0;
@@ -49,3 +50,4 @@ void AHardwareBuffer_release(AHardwareBuffer* /*buffer*/) {}
 int AHardwareBuffer_unlock(AHardwareBuffer* /*buffer*/, int32_t* /*fence*/) {
   return 0;
 }
+}
\ No newline at end of file
diff --git a/nnapi/native/shared_memory.cpp b/nnapi/native/shared_memory.cpp
index 5697d0d6f36..ec422bf2c29 100644
--- a/nnapi/native/shared_memory.cpp
+++ b/nnapi/native/shared_memory.cpp
@@ -12,6 +12,7 @@

 #include <cutils/ashmem.h>

+extern "C" {
 int ASharedMemory_create(const char* name, size_t size) {
   return ashmem_create_region(name, size);
 }
@@ -23,3 +24,4 @@ size_t ASharedMemory_getSize(int fd) {
 int ASharedMemory_setProt(int fd, int prot) {
   return ashmem_set_prot_region(fd, prot);
 }
+}
