From 8fac756ab7c8f508521931d970bcd0c5a393afa5 Mon Sep 17 00:00:00 2001
From: "kidd-kw.chen" <kidd-kw.chen@mediatek.com>
Date: Fri, 5 May 2023 18:25:24 +0800
Subject: [PATCH] add cutom ops

Signed-off-by: kidd-kw.chen <kidd-kw.chen@mediatek.com>
---
 tensorflow/lite/BUILD | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tensorflow/lite/BUILD b/tensorflow/lite/BUILD
index faa089a9fb5..e453a6570a6 100644
--- a/tensorflow/lite/BUILD
+++ b/tensorflow/lite/BUILD
@@ -1275,6 +1275,7 @@ tflite_cc_shared_object(
         ":tflite_exported_symbols.lds",
         ":tflite_version_script.lds",
         "//tensorflow/lite/kernels:builtin_ops_all_linked",
+        "//tensorflow/lite/kernels:custom_ops",
     ],
 )
 
-- 
2.18.0

