From e25f8778516e5d24417b25041dd4f945d9170018 Mon Sep 17 00:00:00 2001
From: "kidd-kw.chen" <kidd-kw.chen@mediatek.com>
Date: Mon, 21 Aug 2023 09:55:42 +0800
Subject: [PATCH 02/10] use syncWait instead of sync_wait

Signed-off-by: Julien STEPHAN <jstephan@baylibre.com>
---
 ArmnnPreparedModel_1_3.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/ArmnnPreparedModel_1_3.cpp b/ArmnnPreparedModel_1_3.cpp
index ceeb3c4..5d57f83 100644
--- a/ArmnnPreparedModel_1_3.cpp
+++ b/ArmnnPreparedModel_1_3.cpp
@@ -15,7 +15,6 @@
 #include <armnn/Types.hpp>
 
 #include <Utils.h>
-#include <android/sync.h>
 #include <log/log.h>
 #include <OperationsUtils.h>
 #include <ExecutionBurstServer.h>
@@ -438,7 +437,7 @@ Return<void> ArmnnPreparedModel_1_3<HalVersion>::executeFenced(const V1_3::Reque
             return Void();
         }
 
-        if (sync_wait(fenceNativeHandle->data[0], -1) < 0)
+        if (syncWait(fenceNativeHandle->data[0], -1) != FenceState::SIGNALED)
         {
             ALOGE("ArmnnPreparedModel_1_3::executeFenced sync fence failed.");
             cb(V1_3::ErrorStatus::GENERAL_FAILURE, hidl_handle(nullptr), nullptr);
-- 
2.18.0

