From 21d285e38ab9218f30d0e5f9b7d7f28b73009bf1 Mon Sep 17 00:00:00 2001
From: Julien STEPHAN <jstephan@baylibre.com>
Date: Mon, 28 Dec 2020 10:31:44 +0000
Subject: [PATCH] add missing cfloat include

Signed-off-by: Julien STEPHAN <jstephan@baylibre.com>
---
 1.2/ArmnnDriverImpl.cpp | 3 ++-
 1.3/ArmnnDriverImpl.cpp | 4 ++--
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/1.2/ArmnnDriverImpl.cpp b/1.2/ArmnnDriverImpl.cpp
index 3274a8a..427050b 100644
--- a/1.2/ArmnnDriverImpl.cpp
+++ b/1.2/ArmnnDriverImpl.cpp
@@ -11,6 +11,7 @@
 #include <armnnDeserializer/IDeserializer.hpp>
 
 #include <log/log.h>
+#include <cfloat>
 #include <sys/stat.h>
 
 namespace
@@ -752,4 +753,4 @@ Return<void> ArmnnDriverImpl::getCapabilities_1_2(const armnn::IRuntimePtr& runt
 }
 
 } // namespace hal_1_2
-} // namespace armnn_driver
\ No newline at end of file
+} // namespace armnn_driver
diff --git a/1.3/ArmnnDriverImpl.cpp b/1.3/ArmnnDriverImpl.cpp
index c8b1d96..f1611c8 100644
--- a/1.3/ArmnnDriverImpl.cpp
+++ b/1.3/ArmnnDriverImpl.cpp
@@ -11,7 +11,7 @@
 #include <armnnDeserializer/IDeserializer.hpp>
 
 #include <log/log.h>
-
+#include <cfloat>
 #include <sys/stat.h>
 
 namespace
@@ -782,4 +782,4 @@ Return<void> ArmnnDriverImpl::getCapabilities_1_3(const armnn::IRuntimePtr& runt
 }
 
 } // namespace hal_1_3
-} // namespace armnn_driver
\ No newline at end of file
+} // namespace armnn_driver
