From f8997e19b716a67c812b4cf1bf56ff84bd04acd4 Mon Sep 17 00:00:00 2001
From: "kidd-kw.chen" <kidd-kw.chen@mediatek.com>
Date: Wed, 12 Jan 2022 17:12:22 +0800
Subject: [PATCH 06/10] android-nn-driver: link libarmnnSerializer

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 4e0b946..b365b42 100644
--- a/Makefile
+++ b/Makefile
@@ -39,7 +39,7 @@ CFLAGS = -Wall -Wextra -std=c++17 -fPIC -fvisibility=default -Wno-inconsistent-m
 
 CXXFLAGS += $(DEFINES) $(INCLUDES) $(CFLAGS)
 
-LIBS = -larmnn
+LIBS = -larmnn -larmnnSerializer
 
 $(LIBNAME): $(OBJECTS)
 	$(CC) -shared -Wl,-soname,$(SONAME) -o $@ $^ -lpthread -lrt -ltensorflowlite ${LIBS}
-- 
2.18.0

