From bf7caf8b24802ed5feb504bab5265a12b0bc6876 Mon Sep 17 00:00:00 2001
From: Suhrid Subramaniam <suhrid.subramaniam@mediatek.com>
Date: Wed, 2 Feb 2022 14:03:50 -0800
Subject: [PATCH 08/10] Add fcntl include depedency for 1.2, 1.3
 armnndriverimpl

Signed-off-by: Suhrid Subramaniam <suhrid.subramaniam@mediatek.com>
---
 1.2/ArmnnDriverImpl.hpp | 3 ++-
 1.3/ArmnnDriverImpl.hpp | 3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/1.2/ArmnnDriverImpl.hpp b/1.2/ArmnnDriverImpl.hpp
index 70f46cb..ad41050 100644
--- a/1.2/ArmnnDriverImpl.hpp
+++ b/1.2/ArmnnDriverImpl.hpp
@@ -6,6 +6,7 @@
 #pragma once
 
 #include <HalInterfaces.h>
+#include <fcntl.h>
 
 #include "../CacheDataHandler.hpp"
 #include "../DriverOptions.hpp"
@@ -60,4 +61,4 @@ public:
 };
 
 } // namespace hal_1_2
-} // namespace armnn_driver
\ No newline at end of file
+} // namespace armnn_driver
diff --git a/1.3/ArmnnDriverImpl.hpp b/1.3/ArmnnDriverImpl.hpp
index a482eda..5329a41 100644
--- a/1.3/ArmnnDriverImpl.hpp
+++ b/1.3/ArmnnDriverImpl.hpp
@@ -6,6 +6,7 @@
 #pragma once
 
 #include <HalInterfaces.h>
+#include <fcntl.h>
 
 #include "../CacheDataHandler.hpp"
 #include "../DriverOptions.hpp"
@@ -59,4 +60,4 @@ public:
 };
 
 } // namespace hal_1_3
-} // namespace armnn_driver
\ No newline at end of file
+} // namespace armnn_driver
-- 
2.18.0

