From 97743ecde3195d070fba86a5105a6417a248fce8 Mon Sep 17 00:00:00 2001
From: "kidd-kw.chen" <kidd-kw.chen@mediatek.com>
Date: Wed, 3 May 2023 13:28:45 +0800
Subject: [PATCH] enable yocto build

Signed-off-by: Julien STEPHAN <jstephan@baylibre.com>
---
 SConstruct | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/SConstruct b/SConstruct
index ca53246834..f40ce401f3 100644
--- a/SConstruct
+++ b/SConstruct
@@ -363,9 +363,7 @@ if 'x86' not in env['arch']:
         elif env['os'] == 'tizen':
             auto_toolchain_prefix = "armv7l-tizen-linux-gnueabi-"
     elif env['estate'] == '64' and 'v8' in env['arch']:
-        if env['os'] == 'linux':
-            auto_toolchain_prefix = "aarch64-linux-gnu-"
-        elif env['os'] == 'bare_metal':
+        if env['os'] == 'bare_metal':
             auto_toolchain_prefix = "aarch64-elf-"
         elif env['os'] == 'android':
             auto_toolchain_prefix = "aarch64-linux-android-"
-- 
2.18.0

