From e0f09944cbe89ca63c420be9e0e51125c4755be8 Mon Sep 17 00:00:00 2001
From: Pedro Tsai <pedro.tsai@mediatek.com>
Date: Fri, 1 Mar 2024 10:31:07 -0800
Subject: [PATCH] Prefer-to-use-libmali-as-the-provider-of-OpenCL

---
 src/core/CL/OpenCL.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/CL/OpenCL.cpp b/src/core/CL/OpenCL.cpp
index 2ebc3274aa..8a2f8ec916 100644
--- a/src/core/CL/OpenCL.cpp
+++ b/src/core/CL/OpenCL.cpp
@@ -48,7 +48,7 @@ CLSymbols &CLSymbols::get()
 
 bool CLSymbols::load_default()
 {
-    static const std::vector<std::string> libraries_filenames{"libOpenCL.so", "libGLES_mali.so", "libmali.so"};
+    static const std::vector<std::string> libraries_filenames{"libmali.so", "libOpenCL.so", "libGLES_mali.so"};
 
     if (_loaded.first)
     {
-- 
2.43.0

