From 7d78dfb75386e88f85e3251f64ace88f4e224902 Mon Sep 17 00:00:00 2001
From: "kidd-kw.chen" <kidd-kw.chen@mediatek.com>
Date: Wed, 3 May 2023 13:29:59 +0800
Subject: [PATCH] Remove unknown variables treated as error

Signed-off-by: Suhrid Subramaniam <suhrid.subramaniam@mediatek.com>
---
 SConstruct | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/SConstruct b/SConstruct
index f40ce401f3..0aa62a7a32 100644
--- a/SConstruct
+++ b/SConstruct
@@ -583,10 +583,3 @@ if env['exceptions']:
         Return()
     SConscript('./tests/SConscript', variant_dir='%s/tests' % build_path, duplicate=0)
 
-# Unknown variables are not allowed
-# Note: we must delay the call of UnknownVariables until after
-# we have applied the Variables object to the construction environment
-unknown = vars.UnknownVariables()
-if unknown:
-    print("Unknown variables: %s" % " ".join(unknown.keys()))
-    Exit(1)
-- 
2.18.0

